### PR TITLE
[Workflow] Added an entered event

### PR DESCRIPTION
**Pull request by Padam87**
| Q             | A
| ------------- | ---
| Branch?       |  3.2
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #20774 (partially) ; #21433
| License       | MIT
| Doc PR        | -

### CODE REVIEW

**Comment by lyrixx** [comment]
Hello @Padam87 

1/ there is already the `enter` event. What is the difference?
2/ What is the use case for this new event?

**Comment by Padam87** [comment]
Hi, I have expained in the issue. The `enter` event is called before the new markings are set, so you can't flush an entity in a listener because of this :)

**Comment by lyrixx** [comment]
Thanks @Padam87 for your time. But  sadly I'm still :-1: about this event. 

We already have many events. More over if you really want to to have your flush outside the controller, you can put it in the marking store. As well it makes more sense since the marking store, is all about the "store", and so the DB.

I need to close this PR as "won't fix" because of the -1 votes from the Symfony Core Team members.

The Workflow component is very recent, so we are very cautious about adding new features to it. In any case, if more users complain about this very same thing, we'll reconsider your proposal for Symfony 3.3. Thanks!

**Comment by lyrixx** [comment]
@Padam87 Could you rebase the PR? Thanks.

File: src/Symfony/Component/Workflow/Workflow.php [hunk 1]
```diff
@@ -238,6 +240,24 @@ private function enter($subject, Transition $transition, Marking $marking)
...
+    {
+        if (null !== $this->dispatcher) {
+            $event = new Event($subject, $marking, $transition);
+
+            $this->dispatcher->dispatch('workflow.entered', $event);
+            $this->dispatcher->dispatch(sprintf('workflow.%s.entered', $this->name), $event);
+        }
+
+        foreach ($transition->getTos() as $place) {
+            $marking->mark($place);
```
Comment (lyrixx) [inline src/Symfony/Component/Workflow/Workflow.php]: The logic is wrong here, the marking is already set in the `enter` method

File: src/Symfony/Component/Workflow/Workflow.php [hunk 2]
```diff
@@ -238,6 +240,24 @@ private function enter($subject, Transition $transition, Marking $marking)
         }
     }
 
+    private function entered($subject, Transition $transition, Marking $marking)
+    {
+        if (null !== $this->dispatcher) {
```
Comment (lyrixx) [inline src/Symfony/Component/Workflow/Workflow.php]: `if (null == $this->dispatcher) { return ; }`

File: src/Symfony/Component/Workflow/Workflow.php (hunk 1 above)
Comment (Padam87) [inline src/Symfony/Component/Workflow/Workflow.php]: Oops, totally missed this, thx

**Comment by lyrixx** [comment]
:+1:

**Review by xabbuh — APPROVED** [review]
:+1:

Status: Reviewed

**Review by lyrixx — APPROVED** [review]

**Comment by lyrixx** [comment]
Thank you @Padam87.

**Comment by fabpot** [comment]
@lyrixx Can you add a note in the component CHANGELOG?

**Comment by lyrixx** [comment]
Yes, I will make a PR.

**Pull request closed without merging**